subsys/shell/modules/devmem_service.c

```diff
@@ -19,6 +19,8 @@
 #include <zephyr/shell/shell.h>
 #include <zephyr/sys/byteorder.h>
 
+extern void getopt_init(void);
```

Comment:
why not include `getopt.h`?

Reply:
I can explain as follows: it seems that `getopt.h` will be included when `CONFIG_SHELL_GETOPT` is enabled.
However, using `getopt` only requires `CONFIG_GETOPT`. Therefore, if we change the configuration to `CONFIG_GETOPT`, there will be no need to directly extern `void getopt_init(void)`.

https://github.com/zephyrproject-rtos/zephyr/blob/28555ae9f4be81f0a1f3ba20e278a8594b372409/include/zephyr/shell/shell.h#L21-L23

Next step if we can expose <getopt.h> by `CONFIG_GETOPT` then we can utilize the following approach instead. 

```
struct getopt_state *getopt_state_get(void);

state = getopt_state_get();
state->optind = 1;
```

Reply:
Oh by the way, it seems cannot pass CI test with (native_sim)

Reply:
I've now managed to get the CI test to pass with the compile switch. The root cause stems from the incompatible implementation between `CONFIG_NATIVE_LIBC`'s `getopt` and `lib\posix\options\getopt\getopt.c`.
So, what could be the long-term solution for this?

Reply:
One possible (preferable?) long term solution should be to use TLS for `optind`, etc. That way, threads will not clobber one another.